packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboard.ts

```diff
@@ -147,31 +147,36 @@ export const useGridKeyboard = (
 
   const handleCellKeyDown = React.useCallback(
     (params: GridCellParams, event: React.KeyboardEvent) => {
-      if (!isGridCellRoot(event.target as HTMLElement)) {
+      // The target is not an element when triggered by a Select inside the cell
+      // See https://github.com/mui-org/material-ui/issues/10534
+      if (event.target instanceof HTMLElement && !isGridCellRoot(event.target)) {
```

Comment:
doesn't work between different documents. 
```suggestion
      if (event.target.nodeType === 1 && !isGridCellRoot(event.target)) {
```

See https://github.com/mui-org/material-ui/blob/33850e52ca32787fbc02b64d3225bac2d451612c/packages/material-ui-utils/src/HTMLElementType.ts#L19